### ISSUE TITLE
Compatibility check to prevent YOG and ZM being run at the same time

### ISSUE THREAD

**Issue by jatkinson1000** [m2lines/CAM-ML#48]
When setting out how YOG would be integrated into CAM it was decided that it should not be an alternative "deep scheme" but instead its own parameterisation.

The recommendation is to set `deep_scheme='off'` when running with YOG, but it has been requested that we add a check that explicitly blocks users from running with `deep_scheme='ZM'` and YOG at the same time.

This is being opened as a separate issue so that it will unblock #35 for merging.
